src/porting.h

```diff
@@ -60,7 +60,16 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 	#include <unistd.h>
 	#include <stdint.h> //for uintptr_t
 
-	#if (defined(linux) || defined(__linux) || defined(__GNU__)) && !defined(_GNU_SOURCE)
+	// Define some platform flags that require a lot of checks
```

Comment:
here too. I agree that its useful, just its unrelated to the change, so would be best to keep it separate.